src/ZkFold/Symbolic/Data/UInt.hs

```diff
@@ -517,13 +532,10 @@ instance (Finite (Zp p), Prime p, KnownNat n, KnownRegisterSize r) => StrictConv
 instance (Arithmetic a, KnownNat n, KnownRegisterSize r, NumberOfBits a <= n) => StrictConv (ArithmeticCircuit a Par1) (UInt n r (ArithmeticCircuit a)) where
     strictConv a = UInt (circuitF $ V.unsafeToVector <$> solve)
         where
-            requiredBits :: Natural
-            requiredBits = (numberOfRegisters @a @n @r -! 1) * (registerSize @a @n @r) + (highRegisterSize @a @n @r)
-
             solve :: MonadBlueprint i a m => m [i]
             solve = do
                 i <- unPar1 <$> runCircuit a
-                bits <- Haskell.reverse <$> expansion requiredBits i
+                bits <- Haskell.reverse <$> expansion (getNatural @n) i -- why count RequiredBits if we can use getNatural @n?
```

Comment:
Good point!
And actually `n` can be bigger than the number of bits which value in `i` can hold so it should be `min n (numberOfBits @a)`, I think